query-planner-js/src/buildQueryPlan.ts

```diff
@@ -688,6 +689,24 @@ function splitFields(
   }
 }
 
+function shouldIncludeToFetchGroup(
+  group: FetchGroup,
+  fields: FieldSet,
+  context: QueryPlanningContext,
+  runtimeParentType: GraphQLObjectType
+  ): Boolean {
+  const { fieldDef } = fields[0];
+  const returnType = getNamedType(fieldDef.type);
+
+  if (isCompositeType(returnType)) {
+    return true;
+  }
+
+  const owningService = context.getOwningService(runtimeParentType, fieldDef);
```

Comment:
We actually need to handle the case where `owningService === undefined` (and `return true`). Otherwise this will create an issue for value types as they don't have an `owningService` and would always be skipped, which we don't want.

An example of this would be if a subgraph has:
```graphql
type Query {
  itf: Itf
}

interface Itf {
  f: String
}

type T1 implements Itf @key(fields: "f") {
  f: String
}

type T2 implements Itf {
  f: String
}
```
With the patch, if you query:
```graphql
{
  itf {
    f
  }
}
```
then the query plan will include `T1` but not `T2` due to this problem.